### PR TITLE
API: jsonref introduction

### PR DESCRIPTION
**Pull request by jma**
⚠️ data importation (setup) and view does not work yet!

* NEW links use now $ref
* REFACTORING rewrite all tests to be invenio-pytest compliant
* REFACTORING put minters and fetchers in api file
* FIX IlsRecord update method
* REFACTORING remove legacy code

Signed-off-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

----------

### To test

- `./scritps/bootstrap`
- `./run-tests.sh`

### CODE REVIEW

File: rero_ils/modules/patrons/utils.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def save_patron(data, record_type, record_class, parent_pid):
 
         patron = record_class.get_patron_by_email(email)
         data['is_patron'] = data.get('is_patron', False)
-        data['is_staff'] = data.get('is_staff', False)
+        data['is_patron'] = data.get('is_patron', False)
```
Comment (BadrAly) [inline rero_ils/modules/patrons/utils.py]: remove duplicate line.

File: tests/data/data.json [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    "$schema": "http://ils.rero.ch/schema/locations/location-v0.0.1.json",
+    "pid": "loc1",
+    "code": "net2-store-base",
+    "name": "Store Base",
+    "is_pickup": true,
+    "library": {
+      "$ref": "http://ils.rero.ch/api/libraries/lib1"
+    }
+  },
+  "circ1": {
```
Comment (BadrAly) [inline tests/data/data.json]: use `cipo` instead?

File: rero_ils/modules/api.py [hunk 3]
```diff
@@ -118,30 +171,69 @@ def get_all_ids(cls, with_deleted=False):
         uuids = [n.object_uuid for n in query]
         return uuids
 
-    def delete(self, force=False, delindex=False):
+    def delete(self, force=False, dbcommit=False, delindex=False):
```
[the remark this answers is not in the document]
Comment (jma) [reply rero_ils/modules/api.py]: Actually this correspond to other create, update default values. I prefer to keep like this to be consistent.

File: rero_ils/modules/circ_policies/views.py [hunk 4]
```diff
@@ -40,20 +42,29 @@
 )
 
 
+def check_permission(fn):
+    """."""
```
[the remark this answers is not in the document]
Comment (jma) [reply rero_ils/modules/circ_policies/views.py]: It will be done at the end.

File: rero_ils/modules/patrons/utils.py (hunk 1 above)
Comment (jma) [reply rero_ils/modules/patrons/utils.py]: will be remove in the next commit

File: rero_ils/modules/documents/jsonresolver.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+"""Organisation resolver."""
+
+
+import jsonresolver
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+
+@jsonresolver.route('/api/documents/<pid>', host='ils.rero.ch')
+def organisation_resolver(pid):
```
Comment (BadrAly) [inline rero_ils/modules/documents/jsonresolver.py]: replace document_resolver by organisation_resolver

**Pull request merged**